modules/table/src/main/java/org/apache/ignite/internal/table/TableImpl.java

```diff
@@ -361,33 +348,21 @@ public void registerSortedIndex(
     public void unregisterIndex(UUID indexId) {
         indexLockerFactories.remove(indexId);
         indexStorageAdapterFactories.remove(indexId);
+
+        CompletableFuture<?> indexToWaitFuture = indexesToWait.remove(indexId);
+
+        if (indexToWaitFuture != null) {
+            indexToWaitFuture.complete(null);
+        }
     }
 
     private void awaitIndexes() {
         // TODO: replace with actual call to ids supplier
-        List<UUID> indexIds = List.of(pkId()); // activeIndexIds.get();
-
         List<CompletableFuture<?>> toWait = new ArrayList<>();
 
-        for (UUID indexId : indexIds) {
-            if (indexLockerFactories.containsKey(indexId) && indexStorageAdapterFactories.containsKey(indexId)) {
```

Comment:
Why did you remove this check?

Reply:
For me, this check does not make sense at the moment.

Reply:
Can you clarify please?

Reply:
If the index is registered, then the future of waiting for the indexes will not be or  completed, so there is no need to check the maps.

Reply:
ok, I've read the code more carefully and looks like this check doesn't make much sense indeed